Populate new courses so that they can be previewed

canstudios-nicolaw opened the pull request:
Fixes #1678 

When you create a course the new course contains a page, article, block and text component. This means that the course can be previewed immediately.

I've put some text in the text component for now which I don't think will be final. We discussed some sort of welcome text in the stand up but didn't specify what it would be. Do we want just a straightforward 'This is an adapt course' type message or something a little more detailed that explains ABC?

canstudios-nicolaw commented on frontend/src/core/project/models/projectContentObjectModel.js:
```diff
@@ -0,0 +1,24 @@
+// LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
```
I don't think this file will be necessary after the refactor PR

dancgray commented:
Block layoutOptions are not populated. Not sure if that matters? Otherwise looks good.

canstudios-nicolaw commented:
I wondered about that to. It didn't seem to do any harm when I took them out, so I left them out. Am happy to put them back in if that's the standard thing though

canstudios-nicolaw commented:
I have added the block layout options to be on the safe side. 

(I'm off next week so I don't want things waiting on me)

lc-thomasberger commented on frontend/src/core/project/views/projectDetailView.js:
```diff
@@ -60,6 +66,82 @@ define(function(require) {
+                      componentModel.save({
+                        _courseId: model.get('_id'),
+                        _parentId: blockModel.get('_id'),
+                        body: 'This is a text component',
```
It would ne nice if we can make this localizable ... or maybe use "Lorem ipsum ..." instead.

taylortom replied:
+1, think that's a good idea @lc-thomasberger

taylortom commented:
Just noticed that various colorPicker assets have changed permissions, was this intentional?

canstudios-louisem commented:
No im not sure how that even happened must have been during the merge

taylortom commented on frontend/src/modules/editor/course/views/editorCourseEditView.js:
```diff
@@ -58,7 +63,84 @@ define(function(require) {
       EditorOriginView.prototype.onSaveError.call(this, null, messageText);
+    },
+
+    /**
```
Nesting reference

taylortom commented on frontend/src/modules/editor/course/views/editorCourseEditView.js:
```diff
@@ -58,7 +63,84 @@ define(function(require) {
+     * @param response
+     * @param options
+     */
+    populateNewCourse: function(model, response, options) {
```
Nesting reference

canstudios-nicolaw commented:
@taylortom I've refactored as per your suggestion (thanks for the code!). We have removed those colourPicker asset changes - sorry we didn't notice those as they didn't show on bitbucket.
